### PR TITLE
Add Conv2D, MaxPool2D classes and features extraction logic

### PR DESCRIPTION
**Pull request by orifmilod**

### CODE REVIEW

**Comment by orifmilod** [comment]
@Farick-m E bekor shidi, hamira review k.

**Review by Farick-m — APPROVED** [review]
I reviewed it to understand what you have done. Ill look into it more when you merge to check what can be improved.

File: mytorch/engine.py [hunk 1]
```diff
@@ -89,6 +90,12 @@ def __truediv__(self, other):
     def __rtruediv__(self, other):
         return other * self**-1
 
+    def __eq__(self, other):
+        return (self.data == other.data if isinstance(other, Value) else other)
+
+    def __gt__(self, other):
```
[the remark this answers is not in the document]
Comment (orifmilod) [reply mytorch/engine.py:97]: It is needed for comparison in here: https://github.com/orifmilod/GigaTorch/pull/8/files#diff-9cef9c99fbb1c4a2ac9e2b83b9f77c8cc279dc02ae05c3ef9d5cd36c5bbb0c25R33-R35

[the remark this answers is not in the document]
Comment (orifmilod) [reply mytorch/engine.py:100]: I have added `__hash__` function overload as it seems like if we implement `__eq__` then the `__hash__ `is set to `None`. I am implementing how the default python implement it, using the `id` of the object

**Pull request merged**